docs/DesignDocs/Attestation_API_Proposal.md

```diff
@@ -0,0 +1,534 @@
+    declared in header file [openenclave/attestation/plugin.h](https://github.com/openenclave/openenclave/blob/v0.9.x/include/openenclave/attestation/plugin.h).
+  - Upon successful evidence generation, `oe_get_evidence()` returns the
+  generated evidence in a dynamically allocated buffer. It also has the option
+  to return a set of endorsements in another dynamically allocated buffer.
+  After processing of the evidence and the endorsements, the caller is
+  responsible to free these buffers.
+  - The `flags` parameter is a legacy inherited from the legacy API
+  `oe_get_report()` to indicate SGX local or remote attestation.
+  It's redundant -- overlapping with the `evidence_format_uuid` parameter
+  which identifies evidence formats with globally unique UUIDs.
+- Free a dynamically allocated evidence buffer.
+  - Function `oe_result_t oe_free_evidence(uint8_t* evidence_buffer)`,
+  declared in header file [openenclave/attestation/plugin.h](https://github.com/openenclave/openenclave/blob/v0.9.x/include/openenclave/attestation/plugin.h)
+- Free a dynamically allocated endorsements buffer.
+  - Function `oe_result_t oe_free_endorsements(uint8_t* endorsements_buffer)`,
+  declared in header file [openenclave/attestation/plugin.h](https://github.com/openenclave/openenclave/blob/v0.9.x/include/openenclave/attestation/plugin.h)
+- Unregister an attester plugin.
+  - Function `oe_unregister_attester(oe_attester_t* plugin)`,
+  declared in header file [openenclave/attestation/plugin.h](https://github.com/openenclave/openenclave/blob/v0.9.x/include/openenclave/attestation/plugin.h).
+
+Use cases for evidence verification:
+
+- Retrieve a single SGX verifier plugin.
+  - Function `oe_verifier_t* oe_sgx_plugin_verifier(void)`, declared in header
+  file [openenclave/attestation/sgx/verifier.h](https://github.com/openenclave/openenclave/blob/v0.9.x/include/openenclave/attestation/sgx/verifier.h).
+  - It returns a single hardcoded SGX verifier plugin for verification of
+  evidences in SGX ECDSA format.
+- Register a verifier plugin.
+  - Function `oe_result_t oe_register_verifier(oe_verifier_t* plugin, const void* config_data, size_t config_data_size)`, declared in header file [openenclave/attestation/plugin.h](https://github.com/openenclave/openenclave/blob/v0.9.x/include/openenclave/attestation/plugin.h).
+  - The verifier plugin returned by `oe_sgx_plugin_verifier()` can be
+  registered with this function.
+  - See notes for the `oe_register_attester()` API function for explanation of
+  the `config_data` parameter.
+- Verify an evidence, optionally using a set of input endorsements and policies.
+  - Function `oe_result_t oe_verify_evidence(const uint8_t* evidence_buffer, size_t evidence_buffer_size, const uint8_t* endorsements_buffer, size_t endorsements_buffer_size, const oe_policy_t* policies, size_t policies_size, oe_claim_t** claims, size_t* claims_length)`,
```

Comment:
Often endorsements will not be supplied across the same protocol as evidence, and the verifier will need to fetch them separately.   Is that intended to be done by the plugin itself underneath this api (i.e., when endorsements buffer is NULL)?

Reply:
Yes.

Reply:
> Current SGX verifier plugin retrieves endorsements underneath the api if endorsements_buffer is null. We expect all verifier plugins to follow the same behavior.

This is in the v0.9 section.